File: fixtures/scale/swift/large/Families/Substitution/SubstitutionBench.swift
// Benchmark entry point for the substitution family; wraps BenchmarkEngine
// with this family's cipher and known-answer vectors.
struct SubstitutionBench {
    private let engine = BenchmarkEngine()

    func measure(iterations: Int) -> BenchSample {
        let cipher = SubstitutionCipher()
        let vectors = SubstitutionVectors.all()
        return engine.sample(cipher, vectors, iterations)
    }
}

File: fixtures/scale/php/large/src/Families/Substitution/SubstitutionBench.php
<?php

namespace CipherLab\Families\Substitution;

use CipherLab\Abstractions\BenchSample;
use CipherLab\Engine\BenchmarkEngine;

// Benchmark entry point for the substitution family; wraps BenchmarkEngine
// with this family's cipher and known-answer vectors.
class SubstitutionBench
{
    private BenchmarkEngine $engine;

    public function __construct()
    {
        $this->engine = new BenchmarkEngine();
    }

    public function measure(int $iterations): BenchSample
    {
        return $this->engine->sample(new SubstitutionCipher(), SubstitutionVectors::all(), $iterations);
    }
}
